index.rst

```diff
@@ -223,6 +221,41 @@ The computing resources needed by the DDS services are separated from those of t
   In the right hand panel, a high level overview of a group of applications running in the same node (CSCs, salkafka producers, etc) sharing the OpenSplice DDS services.
 
 
+.. _DDS-Domain-and-DDS-Partition:
+
+DDS Domain and DDS Partition
+----------------------------
+
+DDS provides two different ways to isolate (or organize) topics; domain and partition.
+
+A DDS domain is a logical network of applications.
+For applications to communicate using DDS they must participate in the same domain.
+Components in different domains are completely isolated from each other.
+For instance, applications running on different domains can have non-compatible versions of the same topic on the same network without interfering with each other.
+
+The DDS partition, on the other hand, is a way of organizing topics on the same domain into different groups.
+The topics are still all part of the same network so all readers and writers on the system must have compatible versions of the topics.
+Nevertheless, if a particular reader would/should never read data from a particular writer, they can be isolated by making sure they are on different partitions.
+The DDS network service still receives the data, which is then filtered out and not made available to the upstream services.
+
+For instance, imagine a system composed of two applications; one that publishes telemetry and subscribes to commands and another that subscribes to telemetry and publishes commands.
+The first application would never read telemetry, only publishes it.
+At the same time, the second application would never read commands, only publishes it.
+
+If all the data is written in the same partition, it means the data reader **on both** applications receives and must handle both telemetry and commands, see left-hand panel on :numref:`fig-partition`.
+On the other hand, if telemetry and commands are placed in two different partitions, the data reader from the first component (that publishes telemetry) would not receive the data the component itself publish (right-hand panel on :numref:`fig-partition`).
+Overall, this feature allows DDS to optimize the data transfer between readers and writers minimizing the load on applications and daemons.
+
+.. figure:: /_static/Partition.png
+   :name: fig-partition
+   :target: ../_images/Partition.png
+   :alt: illustration of the DDS data partitioning
+
+   Illustration of how DDS data partitioning works.
+   Following the discussion in the text, blue arrows represent telemetry data and red arrows commands.
+   On the left hand panel, all data is written to the same partition and therefore, received by all data readers.
+   On the right hand panel data is partitioned so that the reader of the component that writes the data does not receive it back.
+
```

Comment:
At line 375 you introduce shared memory mode, but say nothing about it. A few lines of explanation at the beginning of this section to explain the difference between standard mode and shared memory mode would be very helpful.

Something along the lines of:

In standard mode each DDS application must maintain a cache of DDS messages. If there is a lot of DDS traffic this can be a significant burden on the application. In shared memory mode a DDS daemon maintains the message cache and shares the cache with all DDS applications on the same mode. Note that shared memory mode is not supported in the community edition of ADLink OpenSplice; this feature requires the licensed version.

Reply:
This is explained in the section DDS Communication Architecture above.